x-pack/plugins/enterprise_search/public/applications/app_search/constants/credentials.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const ADMIN = 'admin';
+export const PRIVATE = 'private';
+export const SEARCH = 'search';
```

Comment:
I'd love to get multiple folks' thoughts on where we should store constants architecturally. CC @byronhulcher, @scottybollinger 

We currently have a shared `common/constants.ts`, and it seems to follow we could have `app_search/constants/`, but I'm wondering if it makes more sense to store this directly where it's being used - i.e.,

`enterprise_search/public/applications/app_search/components/credentials/constants.ts`

IMO, this follows the same logic we use for storing styles/assets specific to a view/concern in the same folder, e.g.:

```
|_ components
  |_ some_component
    |- assets/
    |  |- some_component.scss
    |  |_ some_component_bg.png
    |- constants.ts
    |- index.ts
    |- some_component.test.tsx
    |_ some_component.tsx
```

It generally makes working within a certain concern/domain easier when everything related to that concern is within the same folder, and it doesn't particularly (just IMO) make accessing those values from outside that concern any more difficult - it makes sense to go to `components/credentials` to grab credential-related strings.

Thoughts? Concerns?

Reply:
I'm good with that.

Reply:
33cc9d54e4cdb73774c24e3830a2f6c0cea6a976